packages/xsnap/test/xsnap-eof.test.js

```diff
@@ -0,0 +1,289 @@
+  const exitedPKit = makePromiseKit();
+
+  const child =
+    /** @type {import("child_process").ChildProcessWithoutNullStreams} */ (
+      proc.fork(
+        fileURLToPath(
+          new URL(
+            './fixture-xsnap-kill-parent-in-handlecommand.js',
+            import.meta.url,
+          ),
+        ),
+        { stdio: ['ignore', 'pipe', 'pipe', 'ipc'] },
+      )
+    );
+
+  const outP = text(child.stdout);
+  const errP = text(child.stderr);
+
+  child.on('error', err => {
+    exitedPKit.reject(err);
+  });
+
+  child.on('exit', (code, signal) => {
+    exitedPKit.resolve({ code, signal });
+  });
+
+  const [strOut, strErr, { code: nodeExitCode, signal: nodeExitSignal }] =
+    await Promise.all([outP, errP, exitedPKit.promise]);
+
+  t.is(strOut, '', 'stdout must be empty');
+  t.regex(
+    strErr,
+    /Has parent died\?/,
+    'stderr must contain "Has parent died?"',
+  );
+  t.is(nodeExitCode, null, 'exit code must be null');
+  t.is(nodeExitSignal, 'SIGKILL', 'exit signal must be "SIGKILL"');
+});
+
+/**
+ * Launch an xsnap vat that responds to every command by issuing an empty
+ * request of its own, to be received by the provided `handleCommand`.
+ * Returns the vat along with values capturing its surface area (ChildProcess
+ * object and exit promise, command input/output streams, and promises for
+ * stdout/stderr text).
+ *
+ * @param {import('../src/xsnap.js').XSnapOptions['handleCommand']} handleCommand
+ */
+async function spawnReflectiveWorker(handleCommand) {
+  const exitedPKit = makePromiseKit();
+  /** @type {XsnapChildProcess | undefined} */
+  let xsnapProcess;
+
+  /** @type {typeof import('child_process').spawn} */
+  const spawnSpy = (...args) => {
+    // @ts-expect-error overloaded signature
+    const cp = proc.spawn(...args);
+    xsnapProcess = cp;
+    return cp;
+  };
+
+  const io = { spawn: spawnSpy, os: os.type(), fs, tmpName };
+  const vat = await xsnap({
+    ...options(io),
+    handleCommand,
+    stdout: 'pipe',
+    stderr: 'pipe',
+  });
+  await vat.evaluate(
+    'function handleCommand(message) { issueCommand(new Uint8Array().buffer); };',
+  );
+
+  assert(xsnapProcess);
+
+  const toXsnap = xsnapProcess.stdio[3];
+  const fromXsnap = xsnapProcess.stdio[4];
+  const stderrP = text(xsnapProcess.stderr);
+  const stdoutP = text(xsnapProcess.stdout);
+
+  xsnapProcess.on('error', err => {
+    exitedPKit.reject(err);
+  });
+
+  xsnapProcess.on('exit', (code, signal) => {
+    exitedPKit.resolve({ code, signal });
+  });
+
+  return {
+    vat,
+    xsnapProcess,
+    toXsnap,
+    fromXsnap,
+    stdoutP,
+    stderrP,
+    exitedP: exitedPKit.promise,
+  };
+}
+
+/**
+ * @param {ReturnType<spawnReflectiveWorker>} worker
+ * @param {(worker: typeof worker) => Promise<void>} beforeWait
+ * @param {(worker: typeof worker) => Promise<void>} afterWait
+ */
+async function expectTermination(worker, beforeWait, afterWait) {
```

Comment:
This should fix the lint issues:
```suggestion
/**
 * @template {Awaited<ReturnType<spawnReflectiveWorker>>} [W=any]
 * @param {W} worker
 * @param {(worker: W) => Promise<void>} beforeWait
 * @param {(worker: W) => Promise<void>} afterWait
 */
async function expectTermination(worker, beforeWait, afterWait) {
```
or giving a name to the type returned from `spawnReflectiveWorker`:
```suggestion
/**
 * @typedef {Awaited<ReturnType<spawnReflectiveWorker>>} ReflectiveWorker
 */

/**
 * @param {ReflectiveWorker} worker
 * @param {(worker: ReflectiveWorker) => Promise<void>} beforeWait
 * @param {(worker: ReflectiveWorker) => Promise<void>} afterWait
 */
async function expectTermination(worker, beforeWait, afterWait) {
```